plugins/rbac/src/components/RolesList/RolesList.tsx

```diff
@@ -42,6 +45,16 @@ export const RolesList = () => {
     setRoles(searchResults.length);
   };
 
+  React.useEffect(() => {
+    mountedRef.current = true;
+    if (!loading) {
```

Comment:
@debsmita1 Value of `mountedRef` is being changed but not used anywhere. Am I missing something?

Reply:
Right, I didn't realize that only `isMounted` is being used for the `isLoading` value assignment, in which case the below solution should be sufficient. I have confirmed that it works as the same behavior we looked together before @debsmita1 . Unless there's any concern why the local variable `mounted` was used in the first place.

```
  React.useEffect(() => {
    if (!loading) {
      setIsMounted(true);
    }
  }, [loading]);
```

When I tested this solution I found another issue of this component. The total number of roles keeps refreshing every time useRoles fetches data.

https://github.com/janus-idp/backstage-plugins/assets/26255262/1e5eecd8-6769-4121-8a61-aaf385734fa5

I came up a way to make it only refreshes when the data length changes.
```
export const RolesList = () => {
...
const isInitialLoad = React.useRef(true);
...
  React.useEffect(() => {
    if (!loading) {
      setIsMounted(true);  // set isMounted value on loading value changes

      if (isInitialLoad.current) {
        setRoles(data.length);  // update roles only on initial load or onSearchResultsChange
        isInitialLoad.current = false;
      }
    }
  }, [loading, data]);
 
 // use useMemo to avoid recalculation on every render
 const tableTitle = React.useMemo(() => {
    const roleCount = roles ?? data.length;
    return roleCount > 0 ? `All roles (${roleCount})` : 'All roles';
  }, [roles, data.length]);
...
      <Table
        title={tableTitle}  // still gives default value when roles is undefined or null
        ...
      />
...
```
How do you think?

Reply:
Are we using this `useEffect` because the value of `loading` is changing from the hook? In that case I feel it should be handled at the hook level. It should not flip the loading values from `true` -> `false` -> `true` during fetches. It should go from `true` -> `false` only once when the first set of data is loaded and not after that. After that just the data should be updated with new values. Then we can simply say `isLoading=loading` like previously. @debsmita1 WDYT?

```
React.useEffect(() => {
    if (!loading) {
      setIsMounted(true);
    }
  }, [loading]);
```

Reply:
About the count of the roles changing during subsequent fetches, I feel it should also be coming from the hook and we should not be calculating it in the component. The hook should maintain the state for total number of roles and the component should be able to consistently rely on that value.

Reply:
> About the count of the roles changing during subsequent fetches, I feel it should also be coming from the hook and we should not be calculating it in the component. The hook should maintain the state for total number of roles and the component should be able to consistently rely on that value.

@debsmita1 there was another issue pointed out wherein the roles count keeps refreshing between fetches. I have suggested how we can handle it above. LMK if you want to fix it in this PR or have another ticket for that?

Reply:
The role count now stays as is when the API is fetched. The only time it updates is when the search results update